Feature: Add option to indicate that we're not using a patch file

agwells opened the pull request:
Specify a size of "-1" for the main or patch file, to indicate that the app shouldn't check for the presence of that file or complain if it's missing.

agwells committed:
Add option to indicate that we're not using a patch file

Specify a size of "-1" for the main or patch file, to
indicate that the app shouldn't check for the presence
of that file or complain if it's missing.

agamemnus commented:
Good addition -- thanks. Can you please append to the instructions in (4d) in the readme as well?

agwells commented:
Sure, I think I'll do a general cleanup of the README while I'm at it, if you don't mind.

agamemnus commented:
I only recently went through it, but if you have things you would like to change, go ahead.

agwells committed:
Cleanup of README file

... and documenting that you can put "-1" for a file's size
to indicate that you aren't using that file.

agwells commented:
I found the current Readme a little disorganized and hard to read, when I was trying to get started with this plugin. I humbly submit this rewrite similar to the internal documentation I've written up for other devs at my work.

Let me know if I removed anything important. I took out a lot of the stuff about ant and manually editing JSON files, because it's less relevant for Cordova 6.

agamemnus commented:
Great. Thanks. Good work. I looked through it. I just need to make a few edits to it... I'll do so "in a bit".

agamemnus committed:
Update README.md

agamemnus commented on README.md:
```diff
@@ -3,147 +3,214 @@ cordova-plugin-xapkreader (Agamemnus/Flyingsoft Games edition)
+This plugin helps you to do that, by preferring files in `patch` over files in `main`. That is, if you have a file `pics/kitten.jpg` in your `main` and your `patch`, the plugin will use the `kitten.jpg` from `patch` and ignore the one in `main`.
+
+
+### Expansion file version numbers
```
Do we need this section? The plugin already does all of this so isn't this step extraneous? Google Play should almost always (if not always) delete the old version.

agamemnus replied:
I mean the "Expansion file version numbers" section.

agwells replied:
Hm, I actually wouldn't know. I assumed it was important because all the Google support code for APK Expansions asks for the specific version number of the file you want. But as you point out, it also says you will only ever have at most one main file and one patch file available.

I guess you only really need to use the version number & size settings if you want to guard against strange situations where the APK has been updated but the expansion file has not. In which case, sure, we could remove this section (since its prominence will lead people to try to use those settings when they don't strictly need to). Maybe instead just put a link to https://developer.android.com/google/play/expansion-files.html#Filename by the description of the `XAPK_MAIN_VERSION` variable.

agamemnus replied:
Ok. I will make the appropriate changes in a bit. Thanks.

agwells replied:
I took a crack at it myself just now. :)

agwells committed:
Remove unnecessary section about expansion file versions

Most apps will get by fine without using the file version
number settings, so we can remove this section. People who
really want to use that, can read the Google dev docs to
figure out how it works.
